lib/pure/memfiles.nim

```diff
@@ -245,3 +245,35 @@ proc close*(f: var MemFile) =
   
   if error: raiseOSError(lastErr)
 
+type Record* {.unchecked.} = object
```

Comment:
Easy re: the member spelling.  I honestly figured somewhere there was probably already some kind of type better than Record, maybe some kind of Slice or precursor to it, and thought of Record as a placeholder.

As to purpose, it is only to abstract a slice but in terms of a Cstring buffer - both so one could easily filter lines by saying, e.g. 'if rec.Len >3 and memcmp(rec.Beg, "blah", 4): ...' and also to be clear that the memory is unmanaged unlike most other Nim situations.  Fine to add docs, but thought it wise to first get feedback on the type itself.

As to mutability...Well, that is important, but it also "just depends".  If the MemFile is writable then so are the virtual memory image slices returned by the iterator.  Also, that decision is made at a higher level of context -- by the builder of the MemFile.  While 95+% of use cases might be read-only, read-write ones that only modify the file "in-place" are imaginable.  It also bears noting that the lifetime of the pointed-to memory is also managed here by callers as well as the permission.  So, there is implicitly very much an assumption of people keeping track of what constraints exist on the memory.  Factoring the API differently would probably be desired to reflect both kinds of mutability in the types involved.